packages/@react-stately/selection/src/useMultipleSelectionState.ts

```diff
@@ -37,16 +37,26 @@ export interface MultipleSelectionStateProps extends MultipleSelection {
   allowDuplicateSelectionEvents?: boolean
 }
 
+interface MultipleSelectionStateOptions<T> extends MultipleSelectionStateProps {
+  /** The collection of items. */
+  collection?: Collection<Node<T>>
```

Comment:
Will every collection be a collection of nodes? If not, do we have a guarantee that the items in the Collection will have a `index` property? I relied on that to determine what the first selected key is since the order of the selectedKeys prop is not necessarily from first to last. Maybe we can assume that the keys returned by collection.getKeys is sequential?